acceptance/tests/options/config_file/ttls_cached_custom_and_core_fact.rb

```diff
@@ -0,0 +1,126 @@
+test_name 'missing facts should not invalidate cache' do
+  tag 'risk:high'
+
+  require 'facter/acceptance/user_fact_utils'
+  extend Facter::Acceptance::UserFactUtils
+
+  custom_fact_file = 'custom_fact.rb'
+  custom_fact_name = 'my_custom_fact'
+  custom_cache_group = 'my_custom_group'
+  custom_fact_value = 'banana'
+  core_fact_name = 'system_uptime.seconds'
+
+  fact_content = <<-CUSTOM_FACT
+    Facter.add('#{custom_fact_name}') do
+      setcode do
+        "#{custom_fact_value}"
+      end
+    end
+  CUSTOM_FACT
+
+
+  agents.each do |agent|
+    cache_folder = get_cached_facts_dir(agent['platform'], on(agent, facter('kernelmajversion')).stdout.chomp.to_f)
+    fact_dir = agent.tmpdir('facter')
+    env = { 'FACTERLIB' => fact_dir }
+
+    config_dir = get_default_fact_dir(agent['platform'], on(agent, facter('kernelmajversion')).stdout.chomp.to_f)
+    config_file = File.join(config_dir, 'facter.conf')
+
+    config_data = <<~FACTER_CONF
+      facts : {
+        ttls : [
+          { "#{custom_cache_group}" : 3 days }
+        ]
+      }
+
+      fact-groups : {
+        "#{custom_cache_group}" : ["#{custom_fact_name}", "#{core_fact_name}"],
+      }
+    FACTER_CONF
+
+    step "Agent #{agent}: create config file" do
+      agent.mkdir_p(config_dir)
+      create_remote_file(agent, config_file, config_data)
+
+      fact_file = File.join(fact_dir, custom_fact_file)
+      create_remote_file(agent, fact_file, fact_content)
+    end
+
+    teardown do
+      agent.rm_rf(fact_dir)
+      agent.rm_rf("#{cache_folder}/*")
+      agent.rm_rf(config_file)
+    end
+
+    step "should create cache file once for custom_fact" do
+      on(agent, facter(custom_fact_name, environment: env))
+      output = agent.ls_ld(cache_folder + "/*").split("\n")
+
+      step "custom fact value is correct" do
+        cat_output = agent.cat("#{cache_folder}/#{custom_cache_group}")
+        cache = JSON.parse(cat_output)
+        assert_equal(custom_fact_value, cache[custom_fact_name])
+      end
+
+      step "cache file is written for group" do
+        assert_match(/#{custom_cache_group}/, output[0])
+      end
+
+      step "only 1 cache file is created" do
+        assert_equal(output.count, 1)
+      end
+    end
+
+    step "should update cache file once for core fact" do
+      core_value = on(agent, facter(core_fact_name, environment: env)).stdout.strip
+      ls_output = agent.ls_ld(cache_folder + "/*").split("\n")
+
+      step "custom fact value is correct" do
```

Comment:
This step checks core values, the description should be renamed to `core fact value is correct`